addons/mail/static/src/models/attachment_box_view.js

```diff
@@ -1,11 +1,17 @@
 /** @odoo-module **/
 
+import { useComponentToModel } from '@mail/component_hooks/use_component_to_model';
 import { registerModel } from '@mail/model/model_core';
 import { attr, one } from '@mail/model/model_field';
 import { clear } from '@mail/model/model_field_command';
 
 registerModel({
     name: 'AttachmentBoxView',
+    template: 'mail.AttachmentBoxView',
+    templateGetter: 'attachmentBoxView',
+    componentSetup() {
+        useComponentToModel({ fieldName: 'component' });
```

Comment:
How about getting rid of all this verbose stuff and automatically adding a `component` field to the view models instead?

Reply:
That's the intent, but not do this right now in order to not increase scope of PR.